modin/experimental/engines/omnisci_on_ray/io.py

```diff
@@ -288,3 +338,94 @@ def _prepare_pyarrow_usecols(cls, read_csv_kwargs):
+    def _read_csv_check_support(
+        cls,
+        read_csv_kwargs: ReadCsvKwargsType,
+    ) -> Tuple[bool, str]:
+        """
+        Check whatever or not passed parameters are supported by current modin.read_csv
+        implementation.
+
+        Parameters
+        ----------
+        read_csv_kwargs: ReadCsvKwargsType
+                Parameters of read_csv function.
+
+        Returns
+        -------
+        bool:
+            Whatever passed parameters are supported or not.
+        str:
+            Error message that should be raised if user explicitely set `engine="arrow"`.
+        """
+        filepath_or_buffer = read_csv_kwargs.get("filepath_or_buffer", None)
+        header = read_csv_kwargs.get("header", "infer")
+        names = read_csv_kwargs.get("names", None)
+        engine = read_csv_kwargs.get("engine", None)
+        skiprows = read_csv_kwargs.get("skiprows", None)
+        delimiter = read_csv_kwargs.get("delimiter", None)
+
+        if isinstance(filepath_or_buffer, str):
+            file_extension = os.path.splitext(filepath_or_buffer)[1]
+            if (
+                not cls.file_exists(filepath_or_buffer, only_locally=True)
+                or file_extension != ".csv"
+            ):
+                return False, cls.not_implemented_message
+        elif not cls.pathlib_or_pypath(filepath_or_buffer):
+            return False, cls.not_implemented_message
+
+        read_csv_signature = inspect.signature(cls.read_csv)
+        read_csv_default_kwargs = {
+            k: v.default
+            for k, v in read_csv_signature.parameters.items()
+            if v.default is not inspect.Parameter.empty
+        }
+        for arg in cls.unsupported_args:
+            if read_csv_kwargs[arg] != read_csv_default_kwargs[arg]:
+                return False, cls.not_implemented_message
+        if (
+            delimiter is not None and read_csv_kwargs.get("delim_whitespace", False)
+        ) or delimiter == "\n":
+            return False, cls.not_implemented_message
+
+        if names:
+            empty_pandas_df = pandas.read_csv(
+                **dict(
+                    read_csv_kwargs,
+                    nrows=0,
+                    skiprows=None,
+                    skipfooter=0,
+                    usecols=None,
+                    index_col=None,
+                    names=None,
+                    engine=None if engine == "arrow" else engine,
+                ),
+            )
+            columns_number = len(empty_pandas_df.columns)
+            if header not in [None, 0, "infer"]:
+                return (
+                    False,
+                    "read_csv with 'arrow' engine and provided 'names' parameter supports only 0, None and 'infer' header values",
+                )
+            elif columns_number != len(names):
+                # pyarrow.lib.ArrowInvalid: CSV parse error: Expected 1 columns, got 6
+                return False, cls.not_implemented_message
```

Comment:
This is incorrect input provided by a user. `Not implemented` error message doesn't fit here. We probably should raise errors right in this method to use other types of error, e. g. `ValueError`

Reply:
If we will raise `ValueError` in this method, it won't be catched by `try except` statements in the `modin.experimental.engines.omnisci_on_ray.io.read_csv` function and it will cause discrepancy with Pandas read_csv (Pandas handles such cases without exceptions). If specified error message is not enough in that case, we can create custom exception and add it to other catched exceptions [here](https://github.com/modin-project/modin/blob/master/modin/experimental/engines/omnisci_on_ray/io.py#L219)

Reply:
I assumed Pandas would raise an error here. Why can't we follow Pandas's logic here (e.g. truncate names list)?

Reply:
I think we can, but only after conversion from arrow table (after calling `from_arrow` function, we can add additional `NaN` column or combine columns), but it is better to separate this step to another PR.

Reply:
Sure, these cases don't seem to be of high priority.